odrivemotor/src/odriveCAN/odriveCAN.py

```diff
@@ -127,38 +127,34 @@ def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceNam
             self.set_node_id(new_nodeID)
 
     async def set_power(self, power: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        if abs(power) < 0.1:
+            LOGGER.error("Cannot move motor at a power percent that is nearly 0")
         torque = power*self.current_limit*self.torque_constant
         await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
         await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
         await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x01, 'Input_Mode': 0x01})
         await self.send_can_message('Set_Input_Torque', {'Input_Torque': torque})
 
     async def go_for(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        if abs(rpm) < 0.1:
+            LOGGER.error("Cannot move motor at an RPM that is nearly 0")
         rps = rpm / MINUTE_TO_SECOND
+        await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
+        await self.send_can_message('Set_Traj_Vel_Limit', {'Traj_Vel_Limit': abs(rps)})
+        await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
 
-        if revolutions == 0.0:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x02, 'Input_Mode': 0x01})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
-            await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
-            await self.send_can_message('Set_Input_Vel', {'Input_Vel': rps, 'Input_Torque_FF': 0})
-
+        current_position = await self.get_position()
+        goal_position = 0
+        if rpm > 0.0:
+            goal_position = current_position+revolutions+self.offset
+            await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
         else:
-            await self.send_can_message('Set_Controller_Mode', {'Control_Mode': 0x03, 'Input_Mode': 0x05})
-            await self.send_can_message('Set_Traj_Vel_Limit', {'Traj_Vel_Limit': abs(rps)})
-            await self.send_can_message('Set_Axis_State', {'Axis_Requested_State': 0x08})
-            await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
+            goal_position = current_position-revolutions+self.offset
+            await self.send_can_message('Set_Input_Pos', {'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
```

Comment:
What is this if else statement doing now?

Reply:
I think GitHub is putting this in the wrong spot, its actually old lines 153-158, and it's just to set the goal position + or - the current position based on if the rpm is positive or negative